app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -611,8 +612,24 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 refreshCurrentView(false);
                 break;
         }
+        mFormController.setPendingCalloutFormIndex(null);
```

Comment:
This needs to also execute after line 482 where the intent is processed for cancellation.